packages/request/lib/Client.js

```diff
@@ -320,6 +320,11 @@ class Client {
         body
       })
     }
+
+    if (returnResponse) {
```

Comment:
as you only need the headers for this feature, how about making use of the afterResponse hook (line 314)?

Reply:
I think the headers are not available in the hook but I'm not 100% sure. I remember that I tried to use the hook but it was not possible. @holgerkoser may has a better idea regarding the implementation

Reply:
I have added an option `onWarning` to the kube-client that give access to the warning header value without changing the method signatures.